### PR TITLE
JC-11

### PR DESCRIPTION
**Pull request by freesky15**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    <packaging>jar</packaging>
+
+    <properties>
+        <maven.compiler.source>17</maven.compiler.source>
+        <maven.compiler.target>17</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <junit.jupiter.version>5.8.1</junit.jupiter.version>
+        <junit.platform.version>1.8.1</junit.platform.version>
+        <spring.boot.starter.web.version>3.1.2</spring.boot.starter.web.version>
+        <spring.boot.starter.data.jpa.version>3.1.2</spring.boot.starter.data.jpa.version>
```
Comment (saffchen) [inline pom.xml]: Why don't use parent spring-boot dependency?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            <groupId>org.junit.platform</groupId>
+            <artifactId>junit-platform-suite</artifactId>
+            <version>${junit.platform.version}</version>
+            <scope>test</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>2.0.7</version>
```
Comment (saffchen) [inline pom.xml]: Move version to properties

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -0,0 +1,9 @@
+spring.liquibase.enabled=true
```
Comment (saffchen) [inline src/main/resources/application.properties]: Maybe use yaml format?

**Review by saffchen — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (freesky15) [reply pom.xml]: Done!

File: pom.xml (hunk 1 above)
Comment (freesky15) [reply pom.xml]: As far as I know Spring Framework contains 4th version of JUnit by default, isn't it?

File: src/main/resources/application.properties (hunk 3 above)
Comment (freesky15) [reply src/main/resources/application.properties]: Done!

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                  type: jsonb
+                  constraints:
+                    nullable: true
+              - column:
+                  name: state
+                  type: varchar(50)
+                  constraints:
+                    nullable: false
+              - column:
+                  name: departureresulttime
```
Comment (pelmegov) [inline src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: What does the field means: departure to where?

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                  type: timestamp
+                  constraints:
+                    nullable: false
+              - column:
+                  name: result
```
Comment (pelmegov) [inline src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: boolean result looks strange, which kind of result?

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                  type: timestamp
+                  constraints:
+                    nullable: false
+              - column:
+                  name: taskid
+                  type: bigint
+                  constraints:
+                    nullable: false
+              - column:
+                  name: solve
```
Comment (pelmegov) [inline src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: solve means some raw user's solution?

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 4 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: This field means time of sending to the result service(the kafka topic) or back to competition service(if the result is false)

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 5 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: This field show a status e.g. true - the task was solved, false - the task wasn't solved

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 6 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Yes, it is.

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 5 above)
Comment (pelmegov) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Is it required if there is a status field which will provide more details, not only true/false?

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 4 above)
Comment (pelmegov) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: I don't think it's required to handle, especially if you have debezium which will send the event to kafka itself. If you want to handle history of event delivery to kafka, it's better to look at the debezium events table to understand what happened.

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 6 above)
Comment (pelmegov) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Then noun sounds bettern than verb for the field name.

**Comment by pelmegov** [comment]
What about start time of processing and end time?

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 4 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Ok, I see

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 5 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: The status field provides the processing status such as "in progress", "finished" etc. The result field provide the result true == success, false == failed.

File: pom.xml (hunk 1 above)
Comment (freesky15) [reply pom.xml]: Done!

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 4 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Done!

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 6 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Done!

**Review by saffchen — APPROVED** [review]

File: src/main/resources/db/changelog/changeset/01-create-processing-table.yaml (hunk 5 above)
Comment (freesky15) [reply src/main/resources/db/changelog/changeset/01-create-processing-table.yaml]: Done!

**Review by pelmegov — APPROVED** [review]

**Pull request merged**